powerauth-java-server/src/test/java/io/getlime/security/powerauth/app/server/controller/api/PowerAuthControllerTest.java

```diff
@@ -125,4 +459,740 @@ void testUpdateActivation_badRequest() throws Exception {
+        unsupportApplicationVersionRequest.setApplicationId(APPLICATION_ID);
+        unsupportApplicationVersionRequest.setApplicationVersionId(applicationVersionId);
+
+        /* Application version app key created by the SQL script */
+        mockMvc.perform(post("/rest/v3/application/version/unsupport")
+                        .content(wrapInRequestObjectJson(unsupportApplicationVersionRequest))
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.responseObject.applicationVersionId").value(applicationVersionId))
+                .andExpect(jsonPath("$.responseObject.supported").value(false));
+
+
+        final SupportApplicationVersionRequest supportApplicationVersionRequest = new SupportApplicationVersionRequest();
+        supportApplicationVersionRequest.setApplicationId(APPLICATION_ID);
+        supportApplicationVersionRequest.setApplicationVersionId(applicationVersionId);
+
+        /* Application version app key created by the SQL script */
+        mockMvc.perform(post("/rest/v3/application/version/support")
+                        .content(wrapInRequestObjectJson(supportApplicationVersionRequest))
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.responseObject.applicationVersionId").value(applicationVersionId))
+                .andExpect(jsonPath("$.responseObject.supported").value(true));
+
+    }
+
+    /**
+     * Tests the creation, listing, and removal of application integrations.
+     * <p>
+     * This test creates a new application integration, retrieves a list of integrations to verify its creation,
+     * and then removes the created integration, again retrieving the list to verify its removal.
+     *
+     * @throws Exception if the mockMvc.perform operation fails.
+     */
+    @Test
+    void testApplicationIntegration() throws Exception {
+        final String integrationName = UUID.randomUUID().toString();
+        final CreateIntegrationRequest createIntegrationRequest = new CreateIntegrationRequest();
+        createIntegrationRequest.setName(integrationName);
+
+        MvcResult result = mockMvc.perform(post("/rest/v3/integration/create")
+                        .content(wrapInRequestObjectJson(createIntegrationRequest))
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.responseObject.name").value(integrationName))
+                .andReturn();
+
+        final String integrationId = convertMvcResultToObject(result,
```

Comment:
🤔 Isn't it complicated to convert json to POJO? What about using something like `JsonPath.read(result.getResponse().getContentAsString(), "$.id")`